### PR TITLE
Add an `options` prop to `SelectField` as an alternative to `children`

### PR DESCRIPTION
**Pull request by jamesricky**
Note: the behavior of the `options` prop differs from `FinalFormSelect` and is only intended to work with static options.
Use the existing `AsyncSelectField` for dynamic options.

-   Each option must have the `value` and `label` properties. A custom structure is not supported.
-   There are no `getOptionLabel` and `getOptionValue` props. The `label` and `value` properties are used directly.
-   The value stored in the form state is the `value` property, not the whole option object.

```tsx
const options: SelectFieldOption[] = [
    { value: "chocolate", label: "Chocolate" },
    { value: "strawberry", label: "Strawberry" },
    { value: "raspberry", label: "Raspberry", disabled: true },
];

// ...

<SelectField name="flavor" label="Select a flavor" options={options} fullWidth />;
```

---

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)

### CODE REVIEW

**Review by nsams — DISMISSED** [review]

File: packages/admin/admin/src/form/fields/SelectField.tsx [hunk 1]
```diff
@@ -1,6 +1,13 @@
+import { MenuItem } from "@mui/material";
+
 import { Field, FieldProps } from "../Field";
 import { FinalFormSelect, FinalFormSelectProps } from "../FinalFormSelect";
 
+export type SelectFieldOption<Value extends string | number = string | number> = {
+    label: React.ReactNode;
```
Comment (johnnyomair) [inline packages/admin/admin/src/form/fields/SelectField.tsx:7]:
```suggestion
    label: ReactNode;
```

File: packages/admin/admin/src/form/fields/SelectField.tsx [hunk 2]
```diff
@@ -9,19 +16,25 @@ type SelectFieldPropsToExtendFromWithoutChildren<Value extends string | number>
 };
 
 export interface SelectFieldProps<Value extends string | number> extends SelectFieldPropsToExtendFromWithoutChildren<Value> {
-    children: ReturnType<Required<SelectFieldPropsToExtendFrom<Value>>["children"]>;
+    children?: ReturnType<Required<SelectFieldPropsToExtendFrom<Value>>["children"]>;
+    options?: SelectFieldOption[];
```
Comment (johnnyomair) [inline packages/admin/admin/src/form/fields/SelectField.tsx:20]:
```suggestion
    options?: SelectFieldOption<Value>[];
```

File: packages/admin/admin/src/form/fields/SelectField.tsx [hunk 3]
```diff
@@ -9,19 +16,25 @@ type SelectFieldPropsToExtendFromWithoutChildren<Value extends string | number>
...
 
-export function SelectField<Value extends string | number>({ componentsProps = {}, children, ...restProps }: SelectFieldProps<Value>) {
+export function SelectField<Value extends string | number>({ componentsProps = {}, children, options, ...restProps }: SelectFieldProps<Value>) {
     const { finalFormSelect: finalFormSelectProps } = componentsProps;
     return (
         <Field {...restProps}>
             {(props) => (
                 <FinalFormSelect<Value> {...props} {...finalFormSelectProps}>
                     {children}
+                    {options?.map((option) => (
```
Comment (johnnyomair) [inline packages/admin/admin/src/form/fields/SelectField.tsx:33]: It should be either options or children, but not both

**Comment by jamesricky** [comment]
> > Note: the behavior of the options prop differs from FinalFormSelect and is only intended to work with static options.
> 
> Is this really a good idea?

I think yes.

This makes using the `options` prop in `SelectField` a lot easier than it is when using `FinalFormSelect`. 

`FinalFormSelect` does need to support the "complecated" version of the `options` prop, as it's also used by `AsyncSelectField` which requires this more complecated version.

**Review by nsams — DISMISSED** [review]

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Pull request merged**